tasks/interview-corejs.md

```diff
@@ -1,60 +1,278 @@
+  - Know static Object methods
+  - Property flags & descriptors (student is able to set property via Object. defineProperty)
+  - Know how to create iterable objects, Symbol.iterator usage
+
+- #### Object as Hash.
+
+  - Be able to loop through Object keys
+
+- #### Arrays Built-in methods
+
+  - Know how to copy array
+  - Know how to copy array part
+  - Know how to modify array
+
+- #### Arrays Iterating, Sorting, Filtering
+
+  - Know how to sort Array
+  - Be able to custom sorting for Array
+  - Be able to filter Array elements
+  - Know several method how to iterate Array elements
+
+- #### Functional Scope
+
+  - Know global scope and functional scope
+  - Know variables visibility areas
+  - Understand nested scopes and able work with them
+
+- #### Functions Parameters / Arguments
+
+  - Know how to define Function parameters
+  - Know difference between parameters passing by value and by reference
+  - Know how to handle dynamic amount of Function parameters
+
+- #### ECMAScript Intermediate
+
+  - Function default parameters
+  - ECMA script modules
+  - Know how to use spread operator for Function arguments
+  - Be able to compare `arguments` and `rest parameters`
+  - Spread operator for Array
+  - Understand and able to use spread operator for Array concatenation
+    Destructuring assignment
+  - Be able to discover destructuring assignment concept
+  - Understand variables and Function arguments destructuring assignment
+  - String templates
+  - Know how `for..of` loop works
+
+- #### Advanced Functions
+
+  - `this` scope
+  - Understand difference between function and method
+  - Understand how `this` works, realize `this` possible issues
+  - Manage `this` scope
+  - Be able to replace `this` scope
+  - Be able to use `call` and `apply` Function build-in methods
+
+- #### Functional Patterns
+
+  - Immediately invoked functional expression `(IIFE)`
+  - Know IIFE pattern
+  - Callback (Function as argument)
+  - Know callback pattern
+  - Understand callback limitations (callback hell)
+  - Binding, binding one function twice
+  - Know how to bind `this` scope to function
+
+- #### Network requests
+
+  - `Fetch` (with usage)
+  - `XMLHTTPRequest` (concept)
+  - `WebSocket` (concept)
+
+- #### Web components
+
+  - Web components, shadow DOM (concept)
+
+- #### Date & time
+
+  - Date object
+  - Date methods, props
+  - Timezones
+  - Internationalization js (Intl)
+
+- #### Closures Advanced
+
+  - Context (lexical environment)
+  - Understand function creation context (lexical environment)
+  - Be able to explain difference between scope and context
+  - Inner/outer lexical environment
+  - Understand lexical environment traversing mechanism
+  - Understand connection between function and lexical environment
+
+- #### Object Oriented Programming
+
+  - `new` keyword
+  - Understand how `new` keyword works
+  - Function constructor
+  - Know function constructor concept
+  - Know constructor function pattern
```

Comment:
Is it all about **constructor functions**? Maybe it's better to make it shorter?